lib/board.rb

```diff
@@ -3,56 +3,36 @@ class Board
 
   def initialize 
     @cells = {
-      one: ' ', two: ' ', three: ' ',
-      four: ' ', five: ' ', six: ' ',
-      seven: ' ', eight: ' ', nine: ' ',
+      one: '', two: '', three: '',
+      four: '', five: '', six: '',
+      seven: '', eight: '', nine: '',
     }
   end
 
-  def cell_empty?(selected_cell)
-    return true if @cells[selected_cell.to_sym] == ' '
-    false
+  def cell_empty?(cell)
+    (@cells[cell]).empty?
   end
 
-  def game_won?(piece)
-    return true if win?(board_rows, piece) || win?(board_columns, piece) || win?(board_diagonals, piece)
-  end
-
-  def render
-    puts "#{@cells[:one]}|#{@cells[:two]}|#{@cells[:three]}"
-    puts '-----'
-    puts "#{@cells[:four]}|#{@cells[:five]}|#{@cells[:six]}"
-    puts '-----'
-    puts "#{@cells[:seven]}|#{@cells[:eight]}|#{@cells[:nine]}"
+  def set_cell(cell, piece)
+      @cells[cell] = piece
   end
 
-  def win?(win_type, piece)
-    win_type.each_slice(3) do |type|
-      return true if type.all? {|cell| cell == piece }
-    end
-    false
-  end
-
-    def board_rows
-    rows = [
-      @cells[:one], @cells[:two], @cells[:three],
-      @cells[:four], @cells[:five], @cells[:six],
-      @cells[:seven], @cells[:eight], @cells[:nine]
-    ]
+  def game_won?(piece)
+    win?(piece)
   end
 
-  def board_columns
-    columns = [
-      @cells[:one], @cells[:four], @cells[:seven],
-      @cells[:two], @cells[:five], @cells[:eight],
-      @cells[:three], @cells[:six], @cells[:nine]
-    ]
+  def win_helper(x, y, z)
+    !(x).empty? && x == y && y == z
```

Comment:
You can actually take this one step further and do
```ruby
def win_helper(first_cell, second_cell, third_cell)
  !@cells[first_cell].empty? && @cells[first_cell] == @cells[second_cell] && @cells[second_cell] == @cells[third_cell]
end
```

Then down below it would be even more succinct / easier to read
```ruby
def win?
  win_helper(:one, :two, :three) ||
  win_helper(:four, :five, :six) ||
  ...
end
```